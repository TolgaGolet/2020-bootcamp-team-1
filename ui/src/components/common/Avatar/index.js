import React from "react";
import styles from "../Common.module.css";
import Popover from "react-bootstrap/Popover";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";

const Avatar = ({ name, onLogout }) => {
  const letters = name
    .split(" ")
    .map(word => word[0].toUpperCase())
    .slice(0, 2);

  const popover = (
    <Popover id="profile-popover">
      <Popover.Content onClick={onLogout} className={styles.PopoverItem}>
        Log Out
      </Popover.Content>
    </Popover>
  );

  return (
    <OverlayTrigger trigger="click" placement="bottom" overlay={popover}>
      <button
        style={{ outline: "none" }}
        className={
          styles.Avatar +
          " d-flex justify-content-center align-items-center rounded-circle cursor-hover"
        }
      >
        {letters}
      </button>
    </OverlayTrigger>
  );
};

export default Avatar;
