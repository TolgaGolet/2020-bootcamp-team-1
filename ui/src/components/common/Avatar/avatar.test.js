import React from "react";
import { shallow } from "enzyme";
import Avatar from "./";

const OverlayComponent = "OverlayTrigger";

const setup = (props = {}) => {
  const component = shallow(<Avatar {...props} />);
  return component;
};

describe("Avatar Component", () => {
  it("Renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders name's first letter correctly", () => {
    const wrapper = setup({ name: "A" });
    const text = wrapper.find("button").text();
    expect(text).toBe("A");
  });

  it("Renders Popover component correctly and without prop", () => {
    const wrapper = setup();
    const button = wrapper.find("button");
    button.simulate("click");
    const overlayProps = wrapper.find(OverlayComponent).props();
    expect(overlayProps.overlay.props.id).toBe("profile-popover");
  });
});
