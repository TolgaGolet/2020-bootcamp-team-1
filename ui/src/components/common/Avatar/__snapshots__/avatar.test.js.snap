// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component Renders correctly 1`] = `ShallowWrapper {}`;
